modules/apps/change-tracking/change-tracking-rest-impl/src/main/java/com/liferay/change/tracking/rest/internal/resource/v1_0/EntryResourceImpl.java

```diff
@@ -14,17 +14,154 @@
 
 package com.liferay.change.tracking.rest.internal.resource.v1_0;
 
+import com.liferay.change.tracking.definition.CTDefinitionRegistryUtil;
+import com.liferay.change.tracking.engine.CTEngineManager;
+import com.liferay.change.tracking.model.CTCollection;
+import com.liferay.change.tracking.model.CTEntry;
+import com.liferay.change.tracking.rest.dto.v1_0.Entry;
 import com.liferay.change.tracking.rest.resource.v1_0.EntryResource;
+import com.liferay.change.tracking.service.CTCollectionLocalService;
+import com.liferay.change.tracking.service.CTEntryLocalService;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.dao.orm.QueryDefinition;
+import com.liferay.portal.kernel.search.Sort;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.vulcan.pagination.Page;
+import com.liferay.portal.vulcan.pagination.Pagination;
+import com.liferay.portal.vulcan.util.SearchUtil;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
 import org.osgi.service.component.annotations.ServiceScope;
 
 /**
  * @author Máté Thurzó
+ * @author Zoltan Csaszi
  */
 @Component(
 	properties = "OSGI-INF/liferay/rest/v1_0/entry.properties",
 	scope = ServiceScope.PROTOTYPE, service = EntryResource.class
 )
 public class EntryResourceImpl extends BaseEntryResourceImpl {
+
+	@Override
+	public Page<Entry> getCollectionEntriesPage(
+			Long collectionId, String changeTypesFilter,
+			String classNameIdsFilter, Boolean collision, String groupIdsFilter,
+			Integer status, String userIdsFilter, Pagination pagination,
+			Sort[] sorts)
+		throws Exception {
+
+		CTCollection ctCollection = _ctCollectionLocalService.fetchCTCollection(
+			GetterUtil.getLong(collectionId));
+
+		if (ctCollection == null) {
+			throw new IllegalArgumentException(
+				"Unable to get change tacking collection " + collectionId);
+		}
+
+		collision = GetterUtil.getBoolean(collision);
+
+		QueryDefinition<CTEntry> queryDefinition =
+			SearchUtil.getQueryDefinition(CTEntry.class, pagination, sorts);
+
+		if (status != null) {
+			queryDefinition.setStatus(status);
+		}
+		else {
+			queryDefinition.setStatus(WorkflowConstants.STATUS_DRAFT);
+		}
+
+		List<CTEntry> ctEntries = _ctEngineManager.getCTEntries(
```

Comment:
Brian likes this inline (basically if it is only used once, not define a variable), like this: https://github.com/brianchandotcom/liferay-portal/blob/b44c2d166c80b6b6f7738f23f9d85ddc244d6417/modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/KnowledgeBaseFolderResourceImpl.java#L74